### PR TITLE
[SPARK-17680] [SQL] [TEST] Added a Testcase for Verifying Unicode Character Support for Column Names and Comments

### PR DESCRIPTION
**Pull request by gatorsmile**
### What changes were proposed in this pull request?

Spark SQL supports Unicode characters for column names when specified within backticks(`). When the Hive support is enabled, the version of the Hive metastore must be higher than 0.12,  See the JIRA: https://issues.apache.org/jira/browse/HIVE-6013 Hive metastore supports Unicode characters for column names since 0.13. 

In Spark SQL, table comments, and view comments always allow Unicode characters without backticks.

BTW, a separate PR has been submitted for database and table name validation because we do not support Unicode characters in these two cases. 
### How was this patch tested?

N/A

### CODE REVIEW

**Comment by gatorsmile** [comment]
Also need to add a test case for data source tables. Will do it later

**Comment by gatorsmile** [comment]
... Hit a bug in the write path... Need to fix it at first...

**Comment by gatorsmile** [comment]
... Hit one more bug in the write path...

**Comment by gatorsmile** [comment]
retest this please

**Comment by gatorsmile** [comment]
retest this please

**Comment by gatorsmile** [comment]
cc @cloud-fan Could you review this PR? Thanks!

File: sql/hive/src/test/scala/org/apache/spark/sql/hive/execution/HiveDDLSuite.scala [hunk 1]
```diff
@@ -148,6 +148,52 @@ class HiveDDLSuite
...
+    val tabName = "tab1"
+    val viewName = "view1"
+    // scalastyle:off
+    // non ascii characters are not allowed in the source code, so we disable the scalastyle.
+    val colName1 = "和"
+    val colName2 = "尼"
+    val comment = "庙"
+    // scalastyle:on
+    withTable(tabName) {
+      // non ascii characters are not allowed in the source code, so we disable the scalastyle.
```
Comment (cloud-fan) [inline sql/hive/src/test/scala/org/apache/spark/sql/hive/execution/HiveDDLSuite.scala]: remove this line?

**Review by cloud-fan — CHANGES_REQUESTED** [review]

**Comment by cloud-fan** [comment]
LGTM

**Comment by gatorsmile** [comment]
retest this please

**Comment by cloud-fan** [comment]
thanks, merging to master/2.1!

**Pull request closed without merging**